IOPool/Streamer/bin/CalcAdler32.cpp

```diff
@@ -23,16 +24,16 @@ int main(int argc, char* argv[]) {
   std::ifstream::pos_type size = file.tellg();
   file.seekg(0, std::ios::beg);
 
-  boost::shared_array<char> ptr(new char[1024 * 1024]);
+  std::unique_ptr<std::vector<char>> ptr(std::make_unique<std::vector<char>>(1024 * 1024));
```

Comment:
Based on the usage, an array looks more appropriate (`std::vector::data()` leaves a bit icky feeling). Could you replace it with
```suggestion
  auto ptr = std::make_unique<char[]>(1024 * 1024);
```
?

Reply:
Yeah, it looks cleaner this way. We could also use std::array, but I think we would still need to use the data() method.